cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CertUtils.java

```diff
@@ -177,4 +180,122 @@ public static boolean doExistingCertificatesDiffer(Secret current, Secret desire
+
+        // skipping if a volume with same name was already added
+        if (volumeList.stream().noneMatch(v -> v.getName().equals(volumeName))) {
+            volumeList.add(VolumeUtils.createSecretVolume(volumeName, certSecretSource.getSecretName(), isOpenShift));
+        }
+    }
+
+    /**
+     * Creates the volume mounts for Secrets with trusted TLS certificates
+     *
+     * @param volumeMountList       List where the volume mounts will be added
+     * @param trustedCertificates   Trusted certificates for TLS connection
+     * @param tlsVolumeMountPath    Path where the TLS certs should be mounted
+     */
+    public static void createTrustedCertificatesVolumeMounts(List<VolumeMount> volumeMountList, List<CertSecretSource> trustedCertificates, String tlsVolumeMountPath) {
+        createTrustedCertificatesVolumeMounts(volumeMountList, trustedCertificates, tlsVolumeMountPath, null);
+    }
+
+    /**
+     * Creates the volume mounts for Secrets with trusted TLS certificates
+     *
+     * @param volumeMountList       List where the volume mounts will be added
+     * @param trustedCertificates   Trusted certificates for TLS connection
+     * @param tlsVolumeMountPath    Path where the TLS certs should be mounted
+     * @param prefix                Prefix used for the volume names to distinguish when the same Secret is mounted for
+     *                              different purposes (e.g. different cluster connections in MM2)
+     */
+    public static void createTrustedCertificatesVolumeMounts(List<VolumeMount> volumeMountList, List<CertSecretSource> trustedCertificates, String tlsVolumeMountPath, String prefix) {
+        if (trustedCertificates != null && !trustedCertificates.isEmpty()) {
+            for (CertSecretSource certSecretSource : trustedCertificates) {
+                maybeAddTrustedCertificateVolumeMount(volumeMountList, certSecretSource, tlsVolumeMountPath, prefix);
+            }
+        }
+    }
+
+    /**
+     * Creates the VolumeMount used for trusted TLS certificates and adds it to the list. It checks if the Secret where
+     * the certificate is stored has been already used or not and makes sure to mount it only once to avoid duplicates
+     * and naming conflicts.
+     *
+     * @param volumeMountList       List where the volume mount will be added
+     * @param certSecretSource      Represents a certificate inside a Secret
+     * @param tlsVolumeMountPath    Path where the TLS certs should be mounted
+     * @param prefix                Prefix used to generate the volume name
+     */
+    private static void maybeAddTrustedCertificateVolumeMount(List<VolumeMount> volumeMountList, CertSecretSource certSecretSource, String tlsVolumeMountPath, String prefix) {
+        String volumeName = prefix != null ? prefix + '-' + certSecretSource.getSecretName() : certSecretSource.getSecretName();
```

Comment:
Should this line here be extracted to a common method with the adding of the volumes? Since the names need to match (as far as I understand it), won't that mean if we want to change the way we create the volume mount name we would also have to change the volume name?

It may be this will be changed in the future PR that this refactor is preparing for anyway, so not worth doing